oeps/oep-0042-arch-authentication.rst

```diff
@@ -0,0 +1,254 @@
+-----------------------
+
+From the `Wikipedia article on Identity provider`_:
+
+  An Identity Provider, also known as Identity Assertion Provider, can:
+
+  -  provide identifiers for users looking to interact with a system
+
+  -  assert to such a system that such an identifier presented by a user is known to the provider
+
+  -  possibly provide other information about the user that is known to the provider
+
+.. _Wikipedia article on Identity provider: https://en.wikipedia.org/wiki/Identity_provider
+
+Independently Deployable Application (IDA)
+------------------------------------------
+
+An internal edX definition used to describe separate applications that make up a complete Open edX installation. Examples of IDAs are: LMS + Studio (edx-platform), Insights, ECommerce, Credentials, etc.
+
+JSON Web Token (JWT)
+--------------------
+
+From `JWT.io Introduction`_:
+
+  JSON Web Token (JWT) is an open standard (`RFC 7519 <https://tools.ietf.org/html/rfc7519>`__) that defines a compact and self-contained way for securely transmitting information between parties as a JSON object. This information can be verified and trusted because it is digitally signed. JWTs can be signed using a secret (with the HMAC algorithm) or a public/private key pair using RSA.
+
+.. _JWT.io Introduction: https://jwt.io/introduction/
+
+
+OAuth 2.0 (OAuth2)
+------------------
+
+An authorization framework that enables applications to obtain limited access to user accounts on an HTTP service. See a `simplified oauth2 explanation by Aaron Parecki`_ or dig into the full `OAuth2 specification`_.
+
+.. _simplified oauth2 explanation by Aaron Parecki: https://aaronparecki.com/oauth-2-simplified/
+.. _OAuth2 specification: https://oauth.net/2/
+
+OpenID Connect (OIDC)
+---------------------
+
+From `OpenID Connect Discovery 1.0`_ document:
+
+  OpenID Connect 1.0 is a simple identity layer on top of the OAuth 2.0 [RFC6749] protocol. It enables Clients to verify the identity of the End-User based on the authentication performed by an Authorization Server, as well as to obtain basic profile information about the End-User in an interoperable and REST-like manner.
+
+
+.. _OpenID Connect Discovery 1.0: https://openid.net/specs/openid-connect-discovery-1_0.html
+
+Decisions
+=========
+
+There are a number of authentication and authorization related decisions that have been made. This OEP is meant to be updated over time as we gain more information, including links to other related Architectural Decisions Records (ADRs).
+
+Single Identity Provider and OAuth Authorization Server
+-------------------------------------------------------
+
+The LMS will act as the sole identity provider and OAuth authorization server for all other surrounding IDAs. The LMS can provide information about the identity of the user to the other IDAs for account creation purposes. The LMS also provides Single Sign-On (SSO) and Single Logout (SLO) to automatically log in and out IDA users.
```

Comment:
What does `for account creation purposes` refer to here?
The sentence is sufficient to end prior to that phrase.